From b05891b5c51a8bf87143a1c5cb97b0a69447c2fb Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sun, 11 Oct 2020 14:31:15 +0000
Subject: [PATCH 3/6] spl: fit: print error message when get external data
 fails

---
 common/spl/spl_fit.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/common/spl/spl_fit.c b/common/spl/spl_fit.c
index a8bfd388b1..eaee109898 100644
--- a/common/spl/spl_fit.c
+++ b/common/spl/spl_fit.c
@@ -278,8 +278,10 @@ static int spl_load_fit_image(struct spl_load_info *info, ulong sector,
 
 	if (external_data) {
 		/* External data */
-		if (fit_image_get_data_size(fit, node, &len))
+		if (fit_image_get_data_size(fit, node, &len)) {
+			puts("Cannot get external data size\n");
 			return -ENOENT;
+		}
 
 		load_ptr = (load_addr + align_len) & ~align_len;
 		length = len;
@@ -289,8 +291,10 @@ static int spl_load_fit_image(struct spl_load_info *info, ulong sector,
 
 		if (info->read(info,
 			       sector + get_aligned_image_offset(info, offset),
-			       nr_sectors, (void *)load_ptr) != nr_sectors)
+			       nr_sectors, (void *)load_ptr) != nr_sectors) {
+			puts("Cannot read external data\n");
 			return -EIO;
+		}
 
 		debug("External data: dst=%lx, offset=%x, size=%lx\n",
 		      load_ptr, offset, (unsigned long)length);
