From 26504d60f33dbe49164a62f5eda777e6273315c4 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sun, 11 Oct 2020 14:38:10 +0000
Subject: [PATCH 5/6] rockchip: arm64: reserve bl32 memory

---
 arch/arm/mach-rockchip/sdram.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rockchip/sdram.c b/arch/arm/mach-rockchip/sdram.c
index 4c637b7767..a7f8c97ce6 100644
--- a/arch/arm/mach-rockchip/sdram.c
+++ b/arch/arm/mach-rockchip/sdram.c
@@ -42,7 +42,15 @@ int dram_init_banksize(void)
 #ifdef CONFIG_ARM64
 	/* Reserve 0x200000 for ATF bl31 */
 	gd->bd->bi_dram[0].start = 0x200000;
+#if CONFIG_NR_DRAM_BANKS == 2
+	gd->bd->bi_dram[0].size = 0x8400000 - gd->bd->bi_dram[0].start;
+	/* Reserve 32M for OPTEE with TA */
+	gd->bd->bi_dram[1].start = gd->bd->bi_dram[0].start
+				+ gd->bd->bi_dram[0].size + 0x2000000;
+	gd->bd->bi_dram[1].size = top - gd->bd->bi_dram[1].start;
+#else
 	gd->bd->bi_dram[0].size = top - gd->bd->bi_dram[0].start;
+#endif
 #else
 #ifdef CONFIG_SPL_OPTEE
 	struct tos_parameter_t *tos_parameter;
@@ -61,7 +69,7 @@ int dram_init_banksize(void)
 		gd->bd->bi_dram[0].start = CONFIG_SYS_SDRAM_BASE;
 		gd->bd->bi_dram[0].size = 0x8400000;
 		/* Reserve 32M for OPTEE with TA */
-		gd->bd->bi_dram[1].start = CONFIG_SYS_SDRAM_BASE
+		gd->bd->bi_dram[1].start = gd->bd->bi_dram[0].start
 					+ gd->bd->bi_dram[0].size + 0x2000000;
 		gd->bd->bi_dram[1].size = top - gd->bd->bi_dram[1].start;
 	}
